### ISSUE TITLE
Adding a playlist to you playlist do not save in the correct order, it's inverted

### ISSUE THREAD

**Issue by oathzed** [z-huang/music#1225]
### Checklist

- [X] I am able to reproduce the bug with the [latest debug version](https://github.com/z-huang/InnerTune/actions).
- [X] I've checked that there is no open or closed issue about this bug.
- [X] This issue contains only one bug.
- [X] The title of this issue accurately describes the bug.

### Steps to reproduce the bug

1. Have a playlist in YouTube music that is sorted by date added recently
2. Save that playlist to innertune by and have it sorted by date added recently
3. If you add a new song to that playlist it will put it to the bottom

### Expected behavior

Here is a video of what happens, you see all the songs I have are the most recent, so if I add a song, it goes to the least recent. https://imgur.com/a/VvX3D5C

I think when you add the songs from YouTube music to a playlist the sorting gets inverted.

### Actual behavior

Check above video and explanation

### Screenshots/Screen recordings

https://imgur.com/a/VvX3D5C

### Logs

Don't have

### InnerTune version

0.5.3

### Android version

Android 14

### Additional information

Nothing

**Comment by oathzed** [comment]
I think I should of taken a video of the YouTube music playlist import to innertune, because I think that's the actual issue, it brings the sort of the items inverted, recently added goes to the bottom..

**Comment by z-huang** [comment]
I think it's not a bug. You can change ascending or descending by clicking on the arrow button.

**Comment by oathzed** [comment]
> I think it's not a bug. You can change ascending or descending by clicking on the arrow button.

It is a bug. Yes you can change the order fine, but when you change it to descending, it will take the playlist that you've just imported as descending and it will invert it to ascending, so if you add something new to that playlist inside innertune it will add it as descending fine, but all the other songs you've just imported will be ascending.

**Comment by z-huang** [comment]
If you sort by added time ascending, new added song should appear at bottom of the playlist, right?

**Comment by oathzed** [comment]
> If you sort by added time ascending, new added song should appear at bottom of the playlist, right?

Yes they should, but you're sorting your playlist by songs you recently added, you wouldn't sort your playlist by ascending you would sort it by descending, so you can see your new songs you've added first.

When you import a playlist from YouTube music, the order defaults to "custom" it comes in sorted properly, but as soon as you switch it to "date added descending" it will switch the oldest songs to the top of the playlist.

**Comment by z-huang** [comment]
Import a playlist can be viewed as adding each song of the playlist from top to bottom. Therefore, it's normal, right?

**Comment by Nolli20** [comment]
You can't add a sense of the playlist button?

**Comment by oathzed** [comment]
> Import a playlist can be viewed as adding each song of the playlist from top to bottom. Therefore, it's normal, right?

Maybe add them from bottom to top for this to work properly

**Comment by z-huang** [comment]
> Maybe add them from bottom to top for this to work properly

But the create time will be wrong, and the order will be inverted if you sort by added time ascending.

**Comment by oathzed** [comment]
> > Maybe add them from bottom to top for this to work properly
> 
> But the create time will be wrong, and the order will be inverted if you sort by added time ascending.

No it won't, it would be correct if you sorted by added time ascending

**Comment by z-huang** [comment]
Can you just use custom order? You can also change the order as you want.

**Comment by Nolli20** [comment]
> Can you just use custom order? You can also change the order as you want.

It's too long

**Comment by oathzed** [comment]
> Can you just use custom order? You can also change the order as you want.

No because if I use custom order, if I added a new song to the playlist it goes to the bottom. Can you make it go to the top then please?

**Comment by z-huang** [comment]
YouTube Music and Spotify are same with InnerTune: new added song shows in playlist bottom.

**Comment by oathzed** [comment]
> YouTube Music and Spotify are same with InnerTune: new added song shows in playlist bottom.

New added songs to playlists do not go to the bottom in YouTube music, I use the app myself

**Comment by z-huang** [comment]
Proof of my words

https://github.com/user-attachments/assets/8204d908-835c-4adc-a823-c7ef813c8eb9

**Comment by oathzed** [comment]
> Proof of my words
> 
>  Screenrecorder-2024-08-16-10-25-24-303.mp4


https://github.com/user-attachments/assets/1d7995b4-7369-422f-945b-d51b3074f797

And proof of mine lol

**Comment by oathzed** [comment]
My playlists are all sorted "newest first"

**Comment by z-huang** [comment]
Oh, I got what you're saying. Whether you sort by newest first or oldest first, the playlist you added to that playlist always stays the same order. It's like YouTube grouped these songs together with the same added time. Unfortunately, my current database design cannot handle this. Please create another feature request for this, and thanks for your patient to discuss with me :)

**Comment by oathzed** [comment]
> Oh, I got what you're saying. Whether you sort by newest first or oldest first, the playlist you added to that playlist always stays the same order. It's like YouTube grouped these songs together with the same added time. Unfortunately, my current database design cannot handle this. Please create another feature request for this, and thanks for your patient to discuss with me :)

I don't think that's quite it. What's funny is I just sorted my YouTube music playlist to "newest last" then I imported it, and it came in sorted as "newest first" or "date added descending" and when I added something to it it popped up at the top. 

Two problems, I'll have to keep my playlist on YouTube music set to newest last if I ever want to sync it, which obviously isn't ideal. Also another thing I noticed, which would be cool is when I add something to that playlist on inntertune and resync it it will be deleted, it would be cool to keep anything you've added on top on that. But I'll have to create a separate issue for that
